commcare_connect/web/templatetags/avatar.py

```diff
@@ -0,0 +1,54 @@
+from django import template
+
+register = template.Library()
+
+
+def _calculate_initials_internal(name):
+    if not name:
+        # Guest
+        return "GU"
+
+    parts = name.split()
+    initials = ""
+    if len(parts) >= 2 and parts[0] and parts[-1]:
+        initials = (parts[0][0] + parts[-1][0]).upper()
+    elif len(parts) == 1 and parts[0]:
+        if len(parts[0]) >= 2:
+            initials = parts[0][:2].upper()
+        else:
+            initials = parts[0][0].upper()
+    elif name:
+        if len(name) >= 2:
+            initials = name[:2].upper()
+        else:
+            initials = name[0].upper()
+
+    return initials if initials else "GU"
+
+
+@register.inclusion_tag("tailwind/components/avatar.html")
+def user_avatar(user, size="small", color_classes="bg-brand-mango text-white"):
```

Comment:
not blocking but it feels more natural to have the template tag just handle the string part and the rest of the logic in the template itself

Reply:
You mean each time repeat the full avatar HTML and just use `{{ request.user|initials}}`. That will not be DRY, right? 

The way we have it here, it's kind of a component now.